FFmpeg-patches/0003_update_writer_for_avc.patch

```diff
@@ -0,0 +1,94 @@
+diff --git a/libavformat/bmxenc.c b/libavformat/bmxenc.c
+index 89e1af42a0..093d277952 100644
+--- a/libavformat/bmxenc.c
++++ b/libavformat/bmxenc.c
+@@ -22,12 +22,14 @@ static int bmx_writer_init(AVFormatContext *s)
+     struct MxfConfig config;
+     config.clip_type = bmx->clip_type;
+     config.shim_name = bmx->shim_name;
++    config.ard_zdf_hdf = bmx->ard_zdf_hdf;
+     config.frame_rate_num = st->time_base.den;
+     config.frame_rate_den = st->time_base.num;
+     config.partition_size_in_frames = bmx->partition_size_in_frames;
+     config.aes3 = bmx->aes3;
+     config.kag_512 = bmx->kag_512;
+     config.single_pass = bmx->single_pass;
++    config.timecode = bmx->timecode;
+ 
+     bmx->internal_data = create_writer(s->url, &config);
+     return !(bmx->internal_data != NULL);
+@@ -69,6 +72,9 @@ static int bmx_write_header(AVFormatContext *s)
+             default:
+                 break;
+         }
++        if (bmx->essence_video != 0 && st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
++            essence_type = bmx->essence_video;
++        }
+ 
+         av_log(s, AV_LOG_INFO, "Add stream %d\n", i);
+         if(essence_type != UNKNOWN_ESSENCE_TYPE) {
+@@ -78,6 +84,22 @@ static int bmx_write_header(AVFormatContext *s)
+                 bmx_set_quantization_bits(bmx->internal_data, i, st->codecpar->bits_per_coded_sample);
+                 bmx_channel_count(bmx->internal_data, i, st->codecpar->channels);
+             }
++            if(st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
++                bmx_afd(bmx->internal_data, i, bmx->afd);
++                switch(essence_type) {
```

Comment:
double switch case.
I think the best is to remove from here the switch and add a ffmpeg log if return of bmx_avci_header is true to tell AVCI header is set.